src/MizarData/include/MizarData/FrameTrackManager.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_MIZAR_FRAME_TRACK_MANAGER_H_
+#define MIZAR_DATA_MIZAR_FRAME_TRACK_MANAGER_H_
+
+#include <absl/functional/bind_front.h>
+
+#include "ClientData/CaptureData.h"
+#include "MizarData/FrameTrack.h"
+#include "MizarData/MizarData.h"
+#include "MizarData/MizarDataProvider.h"
+
+namespace orbit_mizar_data {
+
+template <typename Data>
+class FrameTrackManagerTmpl {
+  using ScopeId = ::orbit_client_data::ScopeId;
+  using ScopeType = ::orbit_client_data::ScopeType;
+  using PresentEvent = orbit_grpc_protos::PresentEvent;
+
+ public:
+  explicit FrameTrackManagerTmpl(const Data* data) : data_(data) {}
+
+  [[nodiscard]] absl::flat_hash_map<FrameTrackId, FrameTrackInfo> GetFrameTracks() const {
+    const std::vector<ScopeId> scope_ids = GetCaptureData().GetAllProvidedScopeIds();
+    absl::flat_hash_map<FrameTrackId, FrameTrackInfo> result;
+    for (const ScopeId scope_id : scope_ids) {
+      const orbit_client_data::ScopeInfo scope_info = GetCaptureData().GetScopeInfo(scope_id);
+      if (scope_info.GetType() == ScopeType::kDynamicallyInstrumentedFunction ||
+          scope_info.GetType() == ScopeType::kApiScope) {
+        result.try_emplace(FrameTrackId(scope_id), scope_info);
+      }
+    }
+
+    for (auto& [source, unused_events] : data_->source_to_present_events()) {
+      result.try_emplace(FrameTrackId(source), source);
+    }
+    return result;
+  }
+
+  [[nodiscard]] std::vector<FrameStartNs> GetFrameStarts(FrameTrackId id, FrameStartNs min_start,
+                                                         FrameStartNs max_start) const {
+    return Visit(
+        absl::bind_front(&FrameTrackManagerTmpl::GetScopeFrameStarts, this, min_start, max_start),
+        absl::bind_front(&FrameTrackManagerTmpl::GetEtwFrameStarts, this, min_start, max_start),
+        id);
+  }
+
+ private:
+  [[nodiscard]] std::vector<FrameStartNs> GetScopeFrameStarts(FrameStartNs min_start,
+                                                              FrameStartNs max_start,
+                                                              ScopeId scope_id) const {
+    const std::vector<const TimerInfo*> timers =
+        GetCaptureData().GetTimersForScope(scope_id, *min_start, *max_start);
+
+    std::vector<FrameStartNs> result;
+    std::transform(std::begin(timers), std::end(timers), std::back_inserter(result),
+                   [](const TimerInfo* timer) { return FrameStartNs(timer->start()); });
```

Comment:
[No raw loops.](https://www.youtube.com/watch?v=W2tWOdzgXHA)

`std::transform` makes intent immediately clear, unlike for-loop. 

Also, thanks to your question I've just discovered `absl::c_transform` that takes an input collection instead of the two iterators. I'll use it here instead.

Reply:
> Also, I'm never convinced that the lambda is compiled out.

[See the comparison](https://godbolt.org/z/fqva16b57)

The code generated for for-loop and std::transform is identical up to register renaming.

Reply:
Well it turns out the speaker addresses this exact case in the video: ["range based for loops can be very elegant"](https://youtu.be/W2tWOdzgXHA?t=1751). Consider using range based for loop for trivial loops.

Reply:
Yeah. I am aware of this part. "No longer than the composition of two functions with an operator". Here we have two method calls and an explicit constructor call, so I don't see how this fits the guideline. 

Anyway, I see 
```
for (const auto& elem: in) {
  out.push_back(f(elem));
}
```
as a control structure that is repeated in the code obnoxiously often. That's why we've got `std::transform` (and `absl::c_transform`) that gives it a clear name and helps not to repeat ourselves.

Reply:
ok!